mlrun/utils/helpers.py

```diff
@@ -262,8 +263,10 @@ def normalize_name(name):
     # [a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?
     name = re.sub(r"\s+", "-", name)
     if "_" in name:
-        logger.warning(
-            "Replacing underscores (_) with dashes (-). Names with underscore will be deprecated, use dashes instead."
+        warnings.warn(
+            "Names with underscore '_' are about to be deprecated, use dashes '-' instead."
+            "Replacing underscores with dashes.",
+            PendingDeprecationWarning,
```

Comment:
`PendingDeprecationWarning` is more for the developers themselves, it being ignored by default. If this is user facing you would like to use [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning)